scene/gui/file_dialog.cpp

```diff
@@ -414,7 +414,12 @@ void FileDialog::update_file_name() {
 		String filter_str = filters[idx];
 		String file_str = file->get_text();
 		String base_name = file_str.get_basename();
-		file_str = base_name + "." + filter_str.strip_edges().to_lower();
+		Vector<String> filter_substr = filter_str.split(";");
+		if (filter_substr.size() >= 2) {
+			file_str = base_name + "." + filter_substr[0].strip_edges().lstrip("*.").to_lower();
```

Comment:
Why use `lstrip` and not `get_extension` in this case, like for the `== 1` case?

Reply:
I saw it from `EditorFileDialog::update_file_name`, not sure why though. should I change it?
https://github.com/godotengine/godot/blob/96884ef41ecbadd8e76f5e42af5f5cd53a902de0/editor/editor_file_dialog.cpp#L675
@akien-mga

Reply:
Ah that would have been useful to mention from the start, it's easier to review a PR when we know it's tested code from another similar class :)

I think both are equivalent and should give the same output, so I'm not sure with EditorFileDialog should use two different calls. Written by two different contributors it seems and this passed through review: 319840bad1754ce33ec60000df7972e73401860d.

It seems I had actually called it out as not being perfect in https://github.com/godotengine/godot/pull/36699#issuecomment-594417487, but didn't think to suggest just using `get_extension()`.

I'd say you can use `get_extension()` here and in `EditorFileDialog` (and check that it still works as expected of course).

Reply:
Pushed that change myself to unblock that PR two years later ;)